### DISCUSSION TITLE
Onboarding Difficulties

### DISCUSSION THREAD

**Discussion by mklarmann** [twentyhq/twenty#6647] [Unsolvable issues]
There are a couple of issues that I had, trying to onboard the product. I basically stopped now trying, and have to come back later, to see if we can use this product for our company.

I thought it might be valuable to list my problems:

- The gmail importer does not fetch all people / companies. It seems to randomly stopped at some point (after the app being not responsive for some time)
- Importing fields is limited to 2000 entries
- The importer failed with an $data error. And did not import the document I provided.
- The import window just closed randomly
- Icons/descriptions of field properties cannot be changed after they are created (who cares about setting the right icon from the start)
- The activity field (after being synchronized with gmail) does not display all emails. Most are missing.
- How do I merge entries (companies / people) that are twice in the database?

In general, the interface responds fairly sluggish.

I do think you have a really promising product. And you are on the right track, with letting companies define their own fields, and integrate, etc.

So I wish you the best...

**Comment by charlesBochet** [comment]
Hi @mklarmann, thanks for the detailed feedback!

Overall, Twenty is facing bad performances currently as we see workspace number and complexity increase. This is mainly linked to the usage of an underlying library we use to discuss with the database in graphql. @Weiko is working on getting rid of it ; it's a heavy work but we expect it to be done this week or next week

Regarding your specific issues:
- Gmail import has also suffered from performance issues ; we have an ongoing work by @bosiraphael to fix the issue, the fix should land today or tomorrow. We will also display the syncStatus to users to provide a better feedback on the health of the system: https://github.com/twentyhq/twenty/issues/6685
- Import is limited to 2000 entries for performance reasons, I think we could make it higher. How many records do you want to import? 
- The importer failed with an $data error ==> I think the error visible in a tool tip when you hover with your mouse but we can definitely do better. @Bonapara what would we like to see here: here is a typical $data error: Variable "$data" got invalid value "20.10.1997" at "data.geburtstag"; Expected type "DateTime". Invalid date format, expected ISO date string. I think ideally we want to catch all these cases in FE before the user hit the import button. @mklarmann could you tell us your specific error so we can improve the FE behavior?
- @gitstart-twenty could you add a ticket about updating descriptions / icons of custom objects to your existing sprint? (@ijreilly FYI)
- Not sure to understand the 'Activity Field' issue
- "How do I merge entries (companies / people) that are twice in the database?" => unfortunately, we do not support this at the moment

**Reply by ijreilly** [reply]
@gitstart-twenty I created and assigned you to an issue: https://github.com/twentyhq/twenty/issues/6692

**Reply by mklarmann** [reply]
Thanks for the swift response. 

- I wanted to import 4050 items
- The Activity Field, I meant the "Activity Tab" that shows all activities on a company (it misses the passed emails from the gmail import). Yet however it displays manual updates of values, that did not change.

I am sorry, I did not want to create unnecessary stress by posting here.

I can see now the error on importing the data, when looking at the source / hovering over it. There seems to be a fallback for all unspecified columns to ""OpportunityStageEnum" enum. ", that does not make sense?


`Variable "$data" got invalid value "We shall recontact" at "data[0].stage"; Value "We shall recontact" does not exist in "OpportunityStageEnum" enum. Variable "$data" got invalid value "We shall recontact" at "data[1].stage"; Value "We shall recontact" does not exist in "OpportunityStageEnum" enum. Variable "$data" got invalid value "Archived / billed and not recurring" at "data[2].stage"; Value "Archived / billed and not recurring" does not exist in "OpportunityStageEnum" enum. Variable "$data" got invalid value "Closed won - Setup / in operation" at "data[3].stage"; Value "Closed won - Setup / in operation" does not exist in "OpportunityStageEnum" enum. V....
`

**Reply by charlesBochet** [reply]
@mklarmann We have released and deploy a new version of Twenty that fixes the gmail issue (and some performances too)
We have merged a PR: https://github.com/twentyhq/twenty/pull/6338 to ease the import of Select but which is not deployed yet
Here the issue is that you are trying to put your some data that does not match the stage enum into the stage column

**Reply by charlesBochet** [reply]
We are also testing the behavior with 10k records limit to see how the system behave now that we have improved the performances :)

**Comment by BOHEUS** [comment]
Closing this discussion as:
- Gmail issue has been resolved 
- import is now capped at 20000 records
- importer is now working properly
- it's possible to edit icons and descriptions of standard and custom objects
- merging records option has been added to Twenty few months ago
